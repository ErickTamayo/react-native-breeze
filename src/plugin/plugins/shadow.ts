import { PluginFunction, PluginPattern } from "./types";

export type PluginGroups = {
  key: string;
};

export const pattern: PluginPattern = ({ keys }) => {
  return new RegExp(`^shadow-?(?<key>${keys("shadow")})?$`);
};

export const plugin: PluginFunction<PluginGroups> = ({ groups, theme }) => {
  const { key } = groups;
  const value = theme(["shadow", key ?? "default"]);
  return value;
};
