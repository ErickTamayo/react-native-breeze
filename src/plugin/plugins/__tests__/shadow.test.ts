import {
  generateInput,
  shouldEvaluateTheCorrectPatternTest,
  shouldMatchOutputSnapshot,
} from "../../helpers/tests";
import { pattern, plugin } from "../shadow";

describe("shadow", () => {
  // prettier-ignore
  const shouldMatch = generateInput("shadow", ["", "sm", "md", "lg" , "xl", "2xl", "none"]);

  shouldEvaluateTheCorrectPatternTest(pattern, {
    shouldMatch,
    shouldNotMatch: ["shadow-unknown"],
  });

  shouldMatchOutputSnapshot(pattern, plugin, shouldMatch);
});
