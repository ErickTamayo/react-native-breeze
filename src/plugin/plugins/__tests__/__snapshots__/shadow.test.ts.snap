// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shadow Should parse (shadow) correctly 1`] = `
Object {
  "shadowColor": "#000000",
  "shadowOffset": Object {
    "height": 1,
    "width": 0,
  },
  "shadowOpacity": 0.1,
  "shadowRadius": 3,
}
`;

exports[`shadow Should parse (shadow-2xl) correctly 1`] = `
Object {
  "shadowColor": "#000000",
  "shadowOffset": Object {
    "height": 25,
    "width": 0,
  },
  "shadowOpacity": 0.1,
  "shadowRadius": 50,
}
`;

exports[`shadow Should parse (shadow-lg) correctly 1`] = `
Object {
  "shadowColor": "#000000",
  "shadowOffset": Object {
    "height": 10,
    "width": 0,
  },
  "shadowOpacity": 0.1,
  "shadowRadius": 15,
}
`;

exports[`shadow Should parse (shadow-md) correctly 1`] = `
Object {
  "shadowColor": "#000000",
  "shadowOffset": Object {
    "height": 4,
    "width": 0,
  },
  "shadowOpacity": 0.1,
  "shadowRadius": 6,
}
`;

exports[`shadow Should parse (shadow-none) correctly 1`] = `
Object {
  "shadowColor": "#000000",
  "shadowOffset": Object {
    "height": 0,
    "width": 0,
  },
  "shadowOpacity": 0,
  "shadowRadius": 2,
}
`;

exports[`shadow Should parse (shadow-sm) correctly 1`] = `
Object {
  "shadowColor": "#000000",
  "shadowOffset": Object {
    "height": 1,
    "width": 0,
  },
  "shadowOpacity": 0.05,
  "shadowRadius": 2,
}
`;

exports[`shadow Should parse (shadow-xl) correctly 1`] = `
Object {
  "shadowColor": "#000000",
  "shadowOffset": Object {
    "height": 20,
    "width": 0,
  },
  "shadowOpacity": 0.1,
  "shadowRadius": 25,
}
`;
